### ISSUE TITLE
TCP Connection stuck in ESTABLISHED or CLOSE_WAIT

### ISSUE THREAD

**Issue by ospgka** [xmppo/Jabber-Net#81]
On closing a connection the tcp-port won't get released. It got stuck in the CLOSE_WAIT state.
A similar problem occurs on an auth-error. The connection stays in the ESTABLISHED state.

Solution for this problem:
There is an overloaded cunstructor of System.Net.Sockets.NetworkStream that has parameter for setting the ownership of the socket. If it is set to true the stream closes the socket properly.
https://msdn.microsoft.com/en-us/library/te7e60bx(v=vs.100).aspx

Fix for AsyncSocket.cs in bedrock:
> https://github.com/ForNeVeR/Jabber-Net/blob/develop/src/JabberNet/bedrock/net/AsyncSocket.cs
> Line: 617

```
#if __MonoCS__
                    m_sock.Blocking = true;
                    m_stream = new NetworkStream(m_sock, true);
                    m_sock.Blocking = false;
#else
                    m_stream = new NetworkStream(m_sock, true);
#endif
```

**Comment by ForNeVeR** [comment]
Oh, thanks for the report and for the fix! I'll check it and integrate soon.

**Comment by ForNeVeR** [comment]
Thank you, fixed in the `develop` branch.

**Comment by ForNeVeR** [comment]
I've published the [3.0.0-preview1](https://www.nuget.org/packages/jabber-net/3.0.0-preview1) version on NuGet. This version includes your patch and also it's the first release of breaking 3.0.0 branch.
